In [ ]:
password = "few9jiew#&gG(ewqi*&\""

In [ ]:
clientID = "359896540592-10n6b6bn1od7giaoumjhbth6vo4o697k.apps.googleusercontent.com"

In [ ]:
clientSecret = "v_EsiP0RS-zwvK3pFTQkIxLc"

In [ ]:
from utils import data_science_utils

In [ ]:
from utils import models
from utils import sql_utils
from utils import mail_utils

In [ ]:
if False:
    sql_utils.getEngine().execute("delete from newsdb.article_searchesabcd;")

In [ ]:
mail_utils.pipelineEmails(verbose=Verbose.INFO)

Labels:

CHAT
SENT
INBOX
IMPORTANT
TRASH
DRAFT
SPAM
CATEGORY_FORUMS
CATEGORY_UPDATES
CATEGORY_PERSONAL
CATEGORY_PROMOTIONS
CATEGORY_SOCIAL
STARRED
UNREAD

url = "https://www.bbc.com/news/live/world-53775682"

In [ ]:
from utils.data_manager import DataManager
from utils.data_science_utils import *

In [ ]:
search_text_title = "The Brain Implants That Could Change Humanity"
search_text_subtitle = "Brains are talking to computers, and computers to brains. Are our daydreams safe?"

In [ ]:
news_vect = DataManager.getModel(Models.NEWS_VECT)

In [ ]:
weights = []
for word in cleanText(search_text_title).split():
    try:
        index = news_vect[attribute]['tfidf_vectorizer'].get_feature_names().index(word)
        weight = news_vect[attribute]['tfidf_vectorizer'].idf_[index]
        weights = weights.append((word, weight))
    except:
        print(word, "not found")
        pass

In [ ]:
weights

In [ ]:
from utils import translate_utils
from utils import mail_utils
from utils import models
from utils import sql_utils
from utils import data_science_utils

In [ ]:
article = mail_utils.downloadArticle(models.ArticleSearch(search_url='https://www.20min.ch/fr/story/a-nouveau-positif-6-semaines-apres-et-la-2e-infection-est-plus-severe-704431886648'))
article

In [ ]:
article = sql_utils.getArticle('https://www.20min.ch/fr/story/developpement-positif-en-suisse-face-au-covid-19-730995909544')

In [ ]:
articles = sql_utils.getDBSession().query(models.MultiLingualArticle).all()

In [ ]:
articles

In [ ]:
article

In [ ]:
results = data_science_utils.getSimilarArticlesFromText('Coronavirus: Progress in Switzerland in the face of Covid-19, according to Marcel Salathé')

In [ ]:
results.keys()

In [ ]:
results['title']

In [ ]:
from utils import sql_utils
from utils import models
from utils import translate_utils
import re
import sqlalchemy

In [ ]:
rss_sources = sql_utils.getDBSession().query(models.RSSFeed).filter(models.RSSFeed.feed_url.contains('spiegel')).all()

In [ ]:
rss_sources

In [ ]:
source_languages = sql_utils.getDBSession().query(models.Source.language).distinct().all()
source_languages

In [ ]:
spiegel_sources = sql_utils.getDBSession().query(models.Source).filter(sqlalchemy.func.lower(models.Source.source_name).contains('spiegel')).all()
spiegel_sources

In [ ]:
spiegel_sources[0].language = 'de'

In [ ]:
spiegel_articles = sql_utils.getDBSession().query(models.RSSFeed).filter(models.RSSFeed.feed_url.contains('spiegel')).all()

In [ ]:
source = sql_utils.getSource('Spiegel')

In [ ]:
source

In [ ]:
result = sql_utils.getDBSession().query(models.Article).filter(models.Article.source_uuid == str(source.source_uuid)).all()

In [ ]:
result

In [ ]:
multi_lingual_article = models.MultiLingualArticle(
    article_uuid=str(result[2].article_uuid),
    title=result[2].title
)

In [ ]:
sql_utils.insertEntry(multi_lingual_article)

In [ ]:
translation = translate_utils.translateText(removeSpecialChars(result[2].title))

In [ ]:
result[2].title = translation.translated_text

In [ ]:
sql_utils.getDBSession().commit()

In [ ]:
multi_lingual_articles = sql_utils.getDBSession().query(models.MultiLingualArticle).all()

In [ ]:
multi_lingual_article

In [ ]:
def removeSpecialChars(text):
    return re.sub(r'[^ \n:.,!?\-A-Za-z0-9À-ÖØ-öø-ÿЀ-ӿ/]+', '', text)

In [ ]:
def findNumbers(text):
    return re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", text)

In [ ]:
ss = [("apple-12.34 ba33na fanc-14.23e-2yapple+45e5+67.56E+3",
       ['-12.34', '33', '-14.23e-2', '+45e5', '+67.56E+3']),
      ('hello X42 I\'m a Y-32.35 string Z30',
       ['42', '-32.35', '30']),
      ('he33llo 42 I\'m a 32 string -30', 
       ['33', '42', '32', '-30']),
      ('h3110 23 cat 444.4 rabbit 11 2 dog', 
       ['3110', '23', '444.4', '11', '2']),
      ('hello 12 hi 89', 
       ['12', '89']),
      ('4', 
       ['4']),
      ('I like 74,600 commas not,500', 
       ['74,600', '500']),
      ('I like bad math 1+2=.001', 
       ['1', '+2', '.001'])]

for s, r in ss:
    rr = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", s)
    if rr == r:
        print('GOOD')
    else:
        print('WRONG', rr, 'should be', r)

In [ ]:
def dropDoubleSpaces(text):
    l = len(text) + 1
    while l != len(text):
        l = len(text)
        text = text.replace('  ', ' ')
    return text

Find all numbers

transform all of them into text

replace them one by one in the text with find and replace, space around

remove double spaces


In [ ]:
import num2words

In [ ]:
num2words.num2words(64.3)

In [ ]:
sql_utils.getDBSession().commit()

In [ ]:
from utils import translate_utils
from utils import models
from utils import sql_utils
from utils import data_science_utils

In [ ]:
from utils.mail_utils import *

In [75]:
from utils import mail_utils
mail_utils.processFailedArticleSearches()

search_url: https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate
search_url: https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate
search_url: https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate
search_url: https://www.rap-up.com/2020/08/19/travis-scott-announces-tenet-single-the-plan/
search_url: https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate
search_url: https://www.nytimes.com/2020/10/05/health/contact-tracing-white-house.html?action=click&module=Spotlight&pgtype=Homepage
search_url: https://alternativi.fr/les-batteries-des-voitures-electriques-sont-elles-propres/589
ERROR:root:Multiple sources (106) found for article https://alternativi.fr/les-batteries-des-voitures-electriques-sont-elles-propres/589
ERROR:root:Article not found, nor downloaded: <Article Search c66f5f21-b21e-49f0-bed9-79906f2fc53f: https://alternativi.fr/les-batteries-des-voitures-electriques-sont-elles-

14

ERROR:root:Multiple sources (106) found for article https://alternativi.fr/les-batteries-des-voitures-electriques-sont-elles-propres/589

ERROR:root:Article not found, nor downloaded: <Article Search c66f5f21-b21e-49f0-bed9-79906f2fc53f: https://alternativi.fr/les-batteries-des-voitures-electriques-sont-elles-propres/589>

In [72]:
from utils import sql_utils
ucas = sql_utils.getUncompletedArticleSearches()

In [73]:
len(ucas)

15

In [74]:
list(map(lambda x: x.search_url, ucas))

['https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate',
 'https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate',
 'https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate',
 'https://www.rap-up.com/2020/08/19/travis-scott-announces-tenet-single-the-plan/',
 'https://www.foxnews.com/politics/top-5-moments-of-the-first-presidential-debate',
 'https://www.nytimes.com/2020/10/05/health/contact-tracing-white-house.html?action=click&module=Spotlight&pgtype=Homepage',
 'https://alternativi.fr/les-batteries-des-voitures-electriques-sont-elles-propres/589',
 'https://www.lemonde.fr/sciences/article/2019/12/17/comment-la-baleine-est-devenue-le-plus-gros-des-animaux_6023214_1650684.html',
 'https://www.nytimes.com/2020/10/05/health/contact-tracing-white-house.html?action=click&module=Spotlight&pgtype=Homepage',
 'https://www.lemonde.fr/sciences/article/2019/12/17/comment-la-baleine-est-devenue-le-plus-gros-des-animaux_

In [65]:
for u in ucas:
    if u.search_url is None:
        sql_utils.getDBSession().delete(u)
        sql_utils.commitSession()

In [84]:
le_temps = sql_utils.getSource('Le Monde')

In [87]:
le_temps.language = 'fr'

In [88]:
sql_utils.commitSession()